src/main/java/at/hannibal2/skyhanni/features/fishing/trophy/GeyserFishing.kt

```diff
@@ -0,0 +1,82 @@
+package at.hannibal2.skyhanni.features.fishing.trophy
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.IslandType
+import at.hannibal2.skyhanni.events.FishingBobberCastEvent
+import at.hannibal2.skyhanni.events.LorenzRenderWorldEvent
+import at.hannibal2.skyhanni.events.LorenzWorldChangeEvent
+import at.hannibal2.skyhanni.events.ReceiveParticleEvent
+import at.hannibal2.skyhanni.utils.LocationUtils.distanceTo
+import at.hannibal2.skyhanni.utils.LorenzUtils.isInIsland
+import at.hannibal2.skyhanni.utils.LorenzVec
+import at.hannibal2.skyhanni.utils.RenderUtils.drawFilledBoundingBox_nea
+import at.hannibal2.skyhanni.utils.SpecialColour
+import net.minecraft.entity.projectile.EntityFishHook
+import net.minecraft.util.AxisAlignedBB
+import net.minecraft.util.EnumParticleTypes
+import net.minecraftforge.fml.common.eventhandler.EventPriority
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.awt.Color
+
+class GeyserFishing {
+    private val config get() = SkyHanniMod.feature.fishing.trophyFishing.geyserOptions
+
+    private var bobber: EntityFishHook? = null
+    private var geyser: LorenzVec? = null
+    private var geyserBox: AxisAlignedBB? = null
+
+    @SubscribeEvent(priority = EventPriority.LOW, receiveCanceled = true)
+    fun onReceiveParticle(event: ReceiveParticleEvent) {
+        if (!shouldProcessParticles() || event.type != EnumParticleTypes.CLOUD) return
+
+        geyser = event.location
+        val potentialGeyser = geyser ?: return
+
+        geyserBox = AxisAlignedBB(
+            potentialGeyser.x - 2, 118.0 - 0.1, potentialGeyser.z - 2,
+            potentialGeyser.x + 2, 118.0 - 0.09, potentialGeyser.z + 2
+        )
```

Comment:
Not sure why this is set here if it is also set before rendering, maybe just keep it as a variable in the onRenderWorld function instead of having it here